SimplenoteTests/PublishControllerTests.swift

```diff
@@ -0,0 +1,99 @@
+    }
+
+    func testUpdatePublishStateToPublishing() {
+        let note = storage.insertSampleNote(simperiumKey: TestConstants.simperiumKey)
+        publishController.updatePublishState(for: note, to: true) { (state) in
+            XCTAssertEqual(state, .publishing)
+        }
+        XCTAssertTrue(note.published)
+        XCTAssertEqual(note.publishURL, "")
+        XCTAssertNotNil(callbackMap.container[TestConstants.simperiumKey])
+    }
+
+    func testUpdatePublishStateToUnpublishing() {
+        let note = storage.insertSampleNote(simperiumKey: TestConstants.simperiumKey)
+        publishController.updatePublishState(for: note, to: false) { (state) in
+            XCTAssertEqual(state, .unpublishing)
+        }
+        XCTAssertFalse(note.published)
+        XCTAssertEqual(note.publishURL, "")
+        XCTAssertNotNil(callbackMap.container[TestConstants.simperiumKey])
+    }
+
+    func testPublishStateIsChangedToPublishedWhenCalledFromListener() {
+        let note = storage.insertSampleNote(simperiumKey: TestConstants.simperiumKey)
+        var publishState: PublishState = .unpublished
+        publishController.updatePublishState(for: note, to: true) { (state) in
+            publishState = state
+        }
+
+        mockAppDelegate.update(note: note, to: true, with: TestConstants.simperiumKey, with: TestConstants.publishURL)
+
+        XCTAssertEqual(publishState, .published)
+    }
+
+    func testPublishStateIsChangedToUnpublishedWhenCalledFromListener() {
+        let note = storage.insertSampleNote(simperiumKey: TestConstants.simperiumKey)
+        var publishState: PublishState = .published
+        publishController.updatePublishState(for: note, to: true) { (state) in
+            publishState = state
+        }
+
+        mockAppDelegate.update(note: note, to: false, with: TestConstants.simperiumKey, with: TestConstants.publishURL)
+
+        XCTAssertEqual(publishState, .unpublished)
+    }
+
+    func testTimeoutRemovesStoredCallback() {
+        let note = storage.insertSampleNote(simperiumKey: TestConstants.simperiumKey)
+        publishController.updatePublishState(for: note, to: true) { (_) in }
+
+        XCTAssertNotNil(callbackMap.container[TestConstants.simperiumKey])
+
+        timerFactory.timer?.fire()
+
+        XCTAssertNil(callbackMap.container[TestConstants.simperiumKey])
+    }
+}
+
+private struct TestConstants {
+    static let simperiumKey = "ABCDEF123456"
+    static let publishURL = "ABC123"
+}
+
+class MockAppDelegate {
+    let publishController: PublishController
+
+    init(publishController: PublishController) {
+        self.publishController = publishController
+    }
+
+    func update(note: Note, to published: Bool, with key: String, with url: String) {
+        note.publishURL = url
+        note.published = published
+        publishController.didReceiveUpdateFromSimperium(for: key as NSString)
+    }
+}
+
+class MockPublishMap: PublishMap {
```

Comment:
Good to drop this subclass, since it doesn't override / add anything to `PublishMap`